universal-application-tool-0.0.1/app/services/applicant/ApplicantData.java

```diff
@@ -118,6 +119,36 @@ public void putList(Path path, ImmutableList<String> value) {
     }
   }
 
+  /**
+   * Puts the names of the repeated entities at the path. Each element in the JSON array at the path
+   * is a JSON object with an "entity_name" scalar and possibly other nested answers to questions or
+   * repeated entities.
+   *
+   * <p>This should not affect any other data that may already exist for the repeated entities.
+   *
+   * @param path a path to repeated entities list
+   * @param entityNames the names of repeated entities
+   */
+  public void putRepeatedEntities(Path path, ImmutableList<String> entityNames) {
+    if (entityNames.isEmpty()) {
+      put(path, ImmutableList.of());
+    } else {
+      for (int i = 0; i < entityNames.size(); i++) {
+        putString(path.atIndex(i).join(REPEATED_ENTITY_NAME_KEY), entityNames.get(i));
+      }
+    }
+  }
+
+  /**
+   * Deletes the entire repeated entity at the index of the path provided.
+   *
+   * @param path a path to a repeated entities list
+   * @param index the index to delete
+   */
+  public void deleteRepeatedEntity(Path path, int index) {
+    this.jsonData.delete(path.atIndex(index).toString());
+  }
```

Comment:
Question (blocking): what user behavior will use this code path?

Reply:
In the repeater question, applicants will be able to delete a repeated entity by clicking a trash can button.

Reply:
That's going to be tricky to provide a good UX for without making it API-driven. Though maybe for just that feature a small API endpoint and some JS wouldn't be so bad.

Reply:
Thinking about this some more, the API-driven approach would end up being a bit heavier than I was thinking last night.